src/utils/api.js

```diff
@@ -10,14 +10,31 @@ export const getTopics = () => {
 	});
 };
 
-export const getArticlesByTopics = (topic) => {
-	return articlesApi
-		.get(`/articles`, {
-			params: { topic },
-		})
-		.then(({ data }) => {
-			return data.articles;
-		});
+export const getArticles = (topic, filter, order) => {
+	if (topic) {
+		return articlesApi
+			.get(`/articles`, {
+				params: {
+					topic,
+					sort_by: filter,
+					order_by: order,
+				},
+			})
+			.then(({ data }) => {
+				return data.articles;
+			});
+	} else {
+		return articlesApi
+			.get(`/articles`, {
+				params: {
+					sort_by: filter,
+					order_by: order,
+				},
+			})
+			.then(({ data }) => {
+				return data.articles;
+			});
+	}
```

Comment:
I'm quite unsure about just using if statement to make separated request is a good idea. Is there a better way to do with **params**...? 🤔

Reply:
You don't need the if logic here, axios params will just ignore any undefined variable